docs(themes): Make naming convention required and not recommended (#16620)
gatsbyjs/gatsby: docs/docs/themes/conventions.md

## Naming

It's required to prefix themes with `gatsby-theme-`. So if you'd like to name your theme "awesome" you

> can name it `gatsby-theme-awesome` and place that as the `name` key in your `package.json`. Prefixing the themes with `gatsby-theme` allows the gatsby to identify the theme packages and compile the theme code.

Reviewer: ```suggestion
can name it `gatsby-theme-awesome` and place that as the `name` key in your `package.json`. Prefixing themes with `gatsby-theme` enables gatsby in identifying theme packages for compilation.
```
